### PR TITLE
Fix/4938

### LINKED ISSUE #4938: [Nakamoto] getsortition.rs try_handle_request does not actually attempt to return the last winning sortition correctly
This function should be returning the last sortition with the winning commit. However, note the if statement returns none for the last_sortition_ch when the fetched block snapshot does not contain a winning sortition. Later in the signer we attempt to use this non existent consensus hash to find the last winning soritition which will of course always fail. Update getsortition.rs to properly fetch the last winning sortition. This can either be fixed inside getsortition.rs or in the chainstate.
See the following relevant sections:
https://github.com/stacks-network/stacks-core/blob/19f753b173cad722d48873e72bf26a9a27010918/stackslib/src/net/api/getsortition.rs#L216

https://github.com/stacks-network/stacks-core/blob/19f753b173cad722d48873e72bf26a9a27010918/stacks-signer/src/chainstate.rs#L359

### PR DESCRIPTION
**Pull request by jcnelson**
Fixes #4938 by requiring that the last block snapshot with a sortiiton be returned from /v3/sortitions.

### CODE REVIEW

**Comment by jferrant** [comment]
Added a test that enforces that the fetch view sortition check returns a valid last sortition (would terminate early with a "NoSortitionView" error otherwise) and we pass local validation checks to force an RPC call to the node. 

Closes https://github.com/stacks-network/stacks-core/issues/4941 as well.

File: testnet/stacks-node/src/tests/signer/v0.rs [hunk 1]
```diff
@@ -223,45 +259,23 @@ fn block_proposal_rejection() {
...
+    let view = SortitionsView::fetch_view(&signer_test.stacks_client).unwrap();
+    let mut block = NakamotoBlock {
         header: NakamotoBlockHeader::empty(),
         txs: vec![],
     };
+    // We want to force the signer to submit the block to the node for validation
+    // Must set the pox treatment validly and consensus hash validlty to prevent early termination
+    block.header.pox_treatment = BitVec::ones(1).unwrap();
+    block.header.consensus_hash = view.cur_sortition.consensus_hash;
+
```
Comment (kantai) [inline testnet/stacks-node/src/tests/signer/v0.rs:272]: I think this modification should be a second block proposal. That way, the test covers both a rejection by the chainstate module and by the proposal endpoint.

Comment (jferrant) [reply testnet/stacks-node/src/tests/signer/v0.rs:272]: The test above covers this other test case but I will combine the two so its more concise/less duplication!

**Review by kantai — APPROVED** [review]

**Review by obycode — APPROVED** [review]
LGTM

**Pull request merged**